docs/git-guide/gitworkflow.md

```diff
@@ -0,0 +1,207 @@
+# Using Git as you work
+
+## Know what branch you're currently working on 
+
+When working with Git, it's crucial to Know which branch you have currently checked out, Most of the Git commands implicitly operate on the current branch. you can determine the currently checked out branch in serveral ways
+
+One of the most common way is with [git status](https://git-scm.com/docs/git-status)
+
+```
+$ git status
+on branch git-demo
+nothing to commit, working directory clean
+```
+
+Another is with [git branch](https://git-scm.com/docs/git-branch) which will display all local branches, with a star icon next to the current branch:
+
+```
+$ git branch
+* git-demo
+  main
+```
+
+You can see more detailed information about your branches, including remote branches, to do the same use, ```git branch -vva```
+
+```
+$ git branch -vva
+  develop                3ca006a [origin/develop] Updating talawa documentation as new PR merged into talawa:automated-docs
+* docs                   3ca006a Updating talawa documentation as new PR merged into talawa:automated-docs
+  remotes/origin/HEAD    -> origin/develop
+  remotes/origin/develop 3ca006a Updating talawa documentation as new PR merged into talawa:automated-docs
+```
+
+## Keep your fork up to date
+
+You'll want to [keep your fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) up-to-date with changes from Talawa's main repositories
+
+First Make sure you have check out your `develop` branch
+
+```
+$ git checkout develop
+Switched to branch 'develop'
+```
+
+Next is pulling the upstream and [pushing the changes](https://docs.github.com/en/get-started/using-git/pushing-commits-to-a-remote-repository) to your remote fork.Then run `git push`
+
+```
+$ git pull upstream develop
+$ git push origin develop
+```
+
+## Work on a feature branch
+
+Working with git can be organized by creating a branch for each issue or feature. Recall from [how Git is different](https://zulip.readthedocs.io/en/latest/git/the-git-difference.html) that Git is designed for lightweight branching and merging. You can and should create as many branches as you’d like.
+
+First, make sure your `develop` branch is up-to-date with Talawa upstream [see how](#keep-your-fork-up-to-date)
+
+Next from your `develop` branch, create a new branch, though it is recommended to provide a descriptive name that relates to your feature or issue, but you can name it anything
```

Comment:
Change to: Next, from your `develop` branch, create a new branch. We recommended naming your branch a descriptive name relating to your feature or issue.

Reply:
@Biki-das 1) Add a space after branch. and before We   2) I had a typo in my sentence above, please change to "We recommend naming your branch"

Reply:
@Biki-das Add the space after the . and before the word We, for example "...create a new branch. We recommend..."

Reply:
@Biki-das see the comment above.